src/preview.js

```diff
@@ -1,51 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
 import countries from '../assets/countries.json';
 import continentNames from '../assets/continent-names.json';
 import continents from '../assets/continents.json';
 import { getEmojiFlag } from './utils';
-import { __, sprintf } from '@wordpress/i18n';
 
 export default function Preview( { countryCode, relatedPosts } ) {
 	if ( ! countryCode ) return null;
 
 	const emojiFlag = getEmojiFlag( countryCode ),
-	      hasRelatedPosts = relatedPosts?.length > 0;
+		hasRelatedPosts = relatedPosts?.length > 0;
+
+	const displayRelatedPostsCount = () => {
+		if ( ! hasRelatedPosts ) {
+			return __( 'There are no related posts.', 'xwp-country-card' );
+		}
+		if ( relatedPosts.length === 1 ) {
+			return __( 'There is one related post:', 'xwp-country-card' );
+		}
+		return sprintf(
+			/* translators: %d is replaced with the number of related posts */
+			__( 'There are %d related posts:', 'xwp-country-card' ),
+			relatedPosts.length
+		);
+	};
 
 	return (
 		<div className="xwp-country-card">
-			<div className="xwp-country-card__media" data-emoji-flag={ emojiFlag }>
-				<div className="xwp-country-card-flag">
-					{ emojiFlag }
-				</div>
+			<div
+				className="xwp-country-card__media"
+				data-emoji-flag={ emojiFlag }
+			>
+				<div className="xwp-country-card-flag">{ emojiFlag }</div>
 			</div>
 			<h3 className="xwp-country-card__heading">
-				{ __( 'Hello from' ) }
-				{ ' ' }
-				<strong>{ countries[countryCode] }</strong>
-				{ ' ' }
-				(<span className="xwp-country-card__country-code">{ countryCode }</span>),
-				{ ' ' }
-				{ continentNames[continents[countryCode]] }!
+				{ __( 'Hello from' ) }{ ' ' }
+				<strong>{ countries[ countryCode ] }</strong> (
+				<span className="xwp-country-card__country-code">
+					{ countryCode }
+				</span>
+				), { continentNames[ continents[ countryCode ] ] }!
 			</h3>
```

Comment:
For cases when : 
- You have a template string with placeholders for where you inject known html elements or React components.
- You have a translation string where you want to communicate placeholders for html elements to those translating the string.

Then you could consider using `createInterpolateElement`.
Ex : https://make.wordpress.org/core/2020/07/17/introducing-createinterpolateelement/